### ISSUE TITLE
Rename and organise client/src/views 

### ISSUE THREAD

**Issue by jpye-finch** [opencrvs/opencrvs-core#3978]
### Description
- make the naming and folder structure consistent 

### Dev tasks

- [ ] Task 1

**Comment by tareq89** [comment]
For central state management library, can Zustand be a candidate? It is catching up very fast and have hooks like familiar interface.

https://npm-compare.com/@reduxjs/toolkit,zustand,jotai,recoil,valtio/#timeRange=ALL

**Comment by Zangetsu101** [comment]
React Router v6
- Nested routes increasing composability
- Parallel data loading with nested routes
- Easier handling of search params with new hooks
- Better form handling using Web Standards API
- Suspense & error handling support
and lots more: https://reactrouter.com/en/main/start/overview

The other thing about search/route params is that we should consider using some runtime validations(e.g. zod) and proper error handling for incorrect values

**Comment by Zangetsu101** [comment]
Another thing to consider is how we want to handle the offline data which is crucial for us. Is using a library sufficient for us? Are there any use cases that require a more custom solution perhaps?

**Comment by naftis** [comment]
> For central state management library, can Zustand be a candidate? It is catching up very fast and have hooks like familiar interface.
> 
> [npm-compare.com/@reduxjs/toolkit,zustand,jotai,recoil,valtio/#timeRange=ALL](https://npm-compare.com/@reduxjs/toolkit,zustand,jotai,recoil,valtio/#timeRange=ALL)

Yep it's a good option if we need to build/refactor our centralized state management. Here is a interesting comparison about the differences in the state management tools from Jotai's website: https://jotai.org/docs/basics/comparison

Out of them all I think Recoil is something we should skip as it seems stale compared to zustand/jotai.

**Comment by naftis** [comment]
> Another thing to consider is how we want to handle the offline data which is crucial for us. Is using a library sufficient for us? Are there any use cases that require a more custom solution perhaps?

Definitely..

Could we suffice with Apollo doing most of offline caching? If there's something special (like PIN) we need to add to localStorage we could listen to the changes using Jotai https://jotai.org/docs/utilities/storage or Zustand https://docs.pmnd.rs/zustand/integrations/persisting-store-data

This would require refactors like locations refactored being searched from gateway
